includes/polling/bgp-peers.inc.php

```diff
@@ -26,6 +26,28 @@
             $peer_data_check = snmpwalk_cache_multi_oid($device, 'hwBgpPeerEntry', [], 'HUAWEI-BGP-VPN-MIB', 'huawei');
         } elseif ($device['os_group'] == 'cisco') {
             $peer_data_check = snmpwalk_cache_oid($device, 'cbgpPeer2RemoteAs', [], 'CISCO-BGP4-MIB');
```

Comment:
SnmpQuery::context()->walk() handles context a bit better.  And is open for improvements.  Take a look at using that instead of snmpwalk_cache_oid()

Reply:
Hey Tony, thanks for the help so far.
The issue we face is that we're missing info for BGP peers in VRF's for Cisco devices. This call tries to fetch BGP peers in the default VRF, which will rreturn empty. The `if (count($peer_data_check) > 0) {` on line 73 will cause the subsequent code to be skipped, although the Cisco code path doesn't even use $peer_data_check.
To fix this for (only) CIsco devices, we're forcing $peer_data_check to be non-empty. To pass the automated tests I do however need to execute this first ("useless" snmpwalk.

I.e. our current issue isn't snmpwalk_cache_oid vs context()->walk() and more importantly, I'm not quite sure why this code was added in the first place. I tried to figure this out from the git history. It seems to have been added for a specific set of devices. But not sure why it is now intertwined in the code and checked for all devices, even if the code path for those devices doesn't need it. Or in this case even worse: if it causes the polling to fail for those devices.
As such I'm a bit reluctant to tweak the polling code for functionality and code paths I can't test and for which I'm not 100% sure what it's for.

Reply:
bgp code is kind of a nightmare and inter-woven.  If you require the walk for tests to pass, then the data is stored in the db, so it must not be useless.

Reply:
Well useless... it's doesn't seem to be checked by the Cisco code path. The automated tests however do require the data. 
I'm trying to fix the BGP peers in VRF issue while stickting to the mantra "don't break the poller" :-) 
Forcing $peer_data_check to not empty (with dummy data) isn't pretty, but changes the least amount of code to satisfy the check at line 78.

Reply:
But at least it is obvious what  is going on.

If the bgp code ever does get re-written it will be like all other new modules where per-os code lives with the OS.

Reply:
added or clause for cisco devices, used the group for a better link to the code at the top